tsdb/chunks/chunks.go

```diff
@@ -291,24 +314,33 @@ func MergeChunks(a, b chunkenc.Chunk) (*chunkenc.XORChunk, error) {
 			aok = ait.Next()
 			bok = bit.Next()
 		}
-	}
-	for aok {
-		at, av := ait.At()
-		app.Append(at, av)
-		aok = ait.Next()
-	}
-	for bok {
-		bt, bv := bit.At()
-		app.Append(bt, bv)
-		bok = bit.Next()
+		if first {
+			mint = minOf(at, bt)
+			first = false
+		}
+		maxt = minOf(at, bt)
+		numSamples++
+		if numSamples == limit {
+			newMetaArray = append(newMetaArray, Meta{Chunk: newChunk, MinTime: mint, MaxTime: maxt})
+			newChunk = chunkenc.NewXORChunk()
+			app, err = newChunk.Appender()
+			if err != nil {
+				return nil, err
+			}
+			numSamples = 0
+			first = true
+		}
 	}
 	if ait.Err() != nil {
 		return nil, ait.Err()
 	}
 	if bit.Err() != nil {
 		return nil, bit.Err()
 	}
-	return newChunk, nil
+	if newChunk.NumSamples() != 0 {
+		newMetaArray = append(newMetaArray, Meta{Chunk: newChunk, MinTime: mint, MaxTime: maxt})
```

Comment:
We know it is array because we all use modern IDEs, so maybe `newMetas`?

Also it's not array, but slice (: